### DISCUSSION TITLE
How to search a user using his email or phone number.

### DISCUSSION THREAD

**Discussion by bytesagar** [prisma/prisma#10515] [Q&A] [answered]
So I have a user with schema
```prisma
model User {
  id          Int      @id @default(autoincrement())
  email       String   @unique
  firstName   String
  lastName    String
  password    String
  userType    String
  phoneNumber String
  createdAt   DateTime @default(now())
}
```

I want to login this user with his email or phone number.
I tried 
```ts
const login = async (req, res) => {
  const { userEmailPhone, password } = req.body;

  try {
    const user = await prisma.user.findMany({
      where: {
       OR: [
{email:userEmailPhone},
{phoneNumber:userEmailPhone}
]
      },
    });

   
    if (!user) {
      return res.json({ status: "failed", message: "Invalid Credientials" });
    }

    
    const isMatch = await bcrypt.compare(password, user[0]?.password);

    if (!isMatch) {
      return res.json({ status: "failed", message: "Invalid Credientials" });
    }

    res.json({ status: "success", user });
  } catch (error) {
    console.log(error);
    res.json({ status: "failed", message: "Something went wrong" });
  }
};
```
Is this the way to make it work

**Comment by austincrim** [comment, marked as answer]
Hey @bytesagar 👋,

Your `findMany` should return a user if either condition match. You might want to consider making `phoneNumber` a `@unique` field as well.

I don't have enough context about your application to speak to the authentication strategy overall, but the query itself looks good.

**Reply by bytesagar** [reply]
@austincrim 
Thank you for the help. I think this should work with unique email and phone number.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
